### ISSUE TITLE
Docker Explorer Container Files additional support in navigating/opening a file

### ISSUE THREAD

**Issue by melutovich** [microsoft/vscode-docker#4072]
I sometimes want to edit a file within a container, currently, as far as I understand, I have to open one by one each of the parent directories (sometimes 7 or so subdirectories) until I am viewing the file name and then click the open button.  This is a lot of mouse clicking to navigate to the file.

**Comment by bwateratmsft** [comment]
Do you have a suggestion about what could be done differently?

**Comment by melutovich** [comment]
I noticed that the remote host explorer (like Ctrl+o), opens, what appears to be a "dialog" with a text line for inputting a file name, and it defaults to my home directory on the host, and in the dialog below the text line input it shows the files and directories in my home directory.  Would it be possible to have this Docker extension to do something similar within the container[perhaps you can ask the Remote team management if that aspect of Remote extension[the directory/file "dialog"] could either be used when also present along with this Docker extension or could that source code [for the dialog] get included within the Docker extension?

If no [to having a container single level directory explorer in dialog], then perhaps just a text line input for typing an absolute path to a file [without the showing the contexts if the path is to a directory], if the input is to a file then to have a "Ok" button / enter key when pressed opens the file.

If yes [container single level directory explorer in dialog], additionally request in the main Docker Explorer to allow right click on a directory to include an option like "open a file..." which would bring up the above "open dialog" defaulting the text input to the directory that was right clicked.

**Comment by bwateratmsft** [comment]
That's good info, thanks. We've actually asked the remote containers team to give us the ability to use that same dialog but haven't had any luck.

I wonder if a "Recent files" could be useful...

**Comment by melutovich** [comment]
"Recent files" would be useful, but that alone just gives the ability to re-open something already opened; I'd still like at a minimum a text line input to type in the absolute path. [but recently I've found I can at least use the arrow keys to navigate a bit better within the "Files" in the container.  Still not ideal as I have to move my fingers away from the normal keyboard position.]

**Comment by melutovich** [comment]
A few points of response.

I've reviewed that discussion and spent considerable time researching, and if that was the only discussion with them that relates to this topic, then I don't think they have been asked about allowing another Microsoft extension to use (in a private, unexposed way*) the remote extensions "backend" to the simple file dialog.
*I don't know much about vs code extensions but I would think that just as the remote extension is private and not having the source code exposed, perhaps just this aspect of that extension could also be used in this vs code docker extension also in a private not source code exposed way.

I previously asked:

> If no [to having a container single level directory explorer in dialog], then perhaps just a text line input for typing an absolute path to a file [without the showing the contexts if the path is to a directory], if the input is to a file then to have a "Ok" button / enter key when pressed opens the file.

Even the above with the "OK" button always enabled and checking if a file when the "OK" is pressed would be enough.

I noticed https://github.com/microsoft/vscode-azurestaticwebapps/issues/487 mentions currently there is an "input box".

I don't fully follow why the using quick picks (which I assume this extension is currently using for the files in a folder/directory) is not performant for this situation, currently when I have to mouse click, arrow keys to open sub folders is it not the same. ?
